### PR TITLE
[CIS-1105] Take a more defensive approach for user.role

### PR DESCRIPTION
**Pull request by tbarbugli**
Replace force-unwrap with a custom case for user roles.

### CODE REVIEW

**Review by b-onc — DISMISSED** [review]
CHANGELOG please

**Review by nuno-vieira — CHANGES_REQUESTED** [review]
Better to replicate the MemberRole's implementation: https://github.com/GetStream/stream-chat-swift/pull/1288

File: Sources/StreamChat/Models/User.swift [hunk 1]
```diff
@@ -93,16 +93,49 @@ extension ChatUser: Hashable {
...
     
-    /// A user that connected using guest user authentication.
-    case guest
+    public init(stringLiteral value: String) {
+        self.init(rawValue: value)
+    }
     
+    static func custom(_ rawValue: String) -> UserRole {
+        .init(rawValue: rawValue)
+    }
```
Comment (b-onc) [inline Sources/StreamChat/Models/User.swift:109]: Why do we need this?

Comment (tbarbugli) [reply Sources/StreamChat/Models/User.swift:109]: no need indeed :)

**Review by nuno-vieira — APPROVED** [review]
LGTM ✅

**Pull request merged**